processor/resourcedetectionprocessor/config.go

```diff
@@ -32,4 +34,21 @@ type Config struct {
 	// Override indicates whether any existing resource attributes
 	// should be overridden or preserved. Defaults to true.
 	Override bool `mapstructure:"override"`
+	// InternalConfig is a list of settings specific to all detectors
+	DetectorConfigs DetectorConfigs `mapstructure:",squash"`
+}
+
+// DetectorConfigs contains user-specified configurations unique to all individual detectors
+type DetectorConfigs struct {
```

Comment:
What about naming this to DetectorConfig? Even though it's a collection of configs, it's worth to keep it singular to keep consistency. In OpenTelemetry, Config types has a significance so it would help the reader/maintainer in the long term to have consistency.